src/psyclone/tests/nemo/test_files/implicit_do_structures.f90

```diff
@@ -36,14 +36,32 @@ program implicit_do_structures
   implicit none
```

Comment:
AFAICS, this file is only used in the one test file (`domain/nemo/transformations/nemo_arrayrange2loop_trans_test.py`) and therefore, to save on IO and make that test more readable, I suggest inlining this code in that test file as a string. That way the test will be a bit more readable as the various cases being tested are contained in the same file.

Reply:
I do agree. There is a difference in that the get_invoke that was loading the file also raised the PSyIR to NEMO. I purposely not do this step now because it shows that the generic UBOUND() operations in the loop bounds use the appropriate base array. I think it also shows more clearly that this part of the functionality if ready to generalise if needed.

Reply:
Oh good. You're right that we do indeed need to generalise it (at least for the Adjoint work) - #1571.